File: AMS_算法公式组件/code/client/source/YssVisAval/service/IVisAdvAlgoService.cs
using FAST.Common.Service.Pojo.Base;
using FAST.Core.Util;
using FAST.Common.Service.Pojo;
using FAST.Core.Exceptions;
using FAST.Common.Service.Services.Base;
using FAST.Core.Communication.Service;
using FAST.Platform.Workflow.Pojo;
using System;
using System.Collections.Generic;
using System.Text;

using YssVisAval.Pojo.AA;
using YssVisAval.Pojo.SelfAlgorithm;

namespace YssVisAval.service
{
    /// <summary> 
    /// 高级算法服务接口
    /// </summary>
    public interface IVisAdvAlgoService : IServiceBus
    {
        /// <summary>
        /// ddd
        /// </summary>
        /// <param name="customize">customize</param>
        /// <returns>dd</returns>
        string getRealExpression(string customize);
        
        /// <summary>
        /// ddd
        /// </summary>
        /// <param name="funType">funType</param>
        /// <returns>dd</returns>
        string getFunc(string funType);
        
        /// <summary>
        /// ddd
        /// </summary>
        /// <param name="iSNew">判断是否是新算法</param>
        /// <param name="targetStr">targetStr</param>
        /// <returns>dd</returns>
        string checkFormula(string iSNew, string targetStr);
        
        /// <summary>
        /// ddd
        /// </summary>
        /// <param name="keyType">keyType</param>
        /// <returns>ddd</returns>
        string getKeyWords(string keyType);

        /// <summary>
        /// ddd
        /// </summary>
        /// <returns>dd</returns>
        string getAllDate();

        /// <summary>
        /// 取得所有参数
        /// </summary>
        /// <returns>参数列表</returns>
        List<Cls_Parameter> getAllParameters();

        /// <summary>
        /// 取得树列表
        /// </summary>
        /// <param name="code">算法类型</param>
        /// <returns>函数列表</returns>
        string getTreeData(string code);

        /// <summary>
        /// 根据code加载参数和返回值
        /// </summary>
        /// <param name="code">1</param>
        /// <returns>2</returns>
        string getDataAPIByCode(string code);

        /// <summary>
        /// 根据参数的code加载控件
        /// </summary>
        /// <param name="code">关键字</param>
        /// <returns>下拉框信息</returns>
        Dictionary<string, string> getParasByCode(string code);

        /// <summary>
        /// 根据算法代码查找所有算法信息及映射关系
        /// </summary>
        /// <param name="algoCode">algoCode</param>
        /// <returns>Dictionary</returns>
        Dictionary<string, string> getAlgoMsg(string algoCode);

        /// <summary>
        /// 根据算法代码查出所有算法映射关系信息
        /// </summary>
        /// <param name="code">algoCode</param>
        /// <returns>string</returns>
        string getAlgoRTF(string code);

        /// <summary>
        /// 找到最大标记
        /// </summary>
        /// <param name="code">code</param>
        /// <returns>string</returns>
        string getMaxSign(string code);

        /// <summary>
        /// 获取A去树结构
        /// </summary>
        /// <param name="paraDict">paraDict</param>
        /// <returns>QueryRes</returns>
        QueryRes getTreeViewData(Dictionary<string, string> paraDict);

        /// <summary>
        /// 获取A区树所有数据
        /// </summary>
        /// <returns>json</returns>
        string getAllDataTree_A();

        /// <summary>
        /// 参考算法数据
        /// </summary>
        /// <returns>string</returns>
        string getAlgos();

        /// <summary>
        /// 根据code获取算法描述
        /// </summary>
        /// <param name="code">code</param>
        /// <returns>s</returns>
        string getAlgoDesc(string code);

        /// <summary>
        /// 根据算法代码查找算法信息
        /// </summary>
        /// <param name="code">code</param>
        /// <returns>BasePojo</returns>
        BasePojo getAlgoByCode(string code);

        /// <summary>
        /// 根据函数代码到后台查询规则
        /// </summary>
        /// <param name="code">函数代码</param>
        /// <returns>规则集合</returns>
        Dictionary<string, bool> getRuleByFunCode(string code);

        /// <summary>
        /// 根据正则表达式解析合法的函数表达式
        /// </summary>
        /// <param name="fun">表达式串</param>
        /// <returns>函数集合</returns>
        List<string> FunRegex(string fun);
        
        /// <summary>
        /// 预览
        /// </summary>
        /// <param name="msg">中文算法</param>
        /// <returns>翻译后文本</returns>
        string preview(string msg);

        /// <summary>
        /// 验证新旧算法
        /// </summary>
        /// <param name="code">code</param>
        /// <returns>string</returns>
        string isNewAlgo(string code);

        /// <summary>
        /// 根据关键字获取保留位数下拉框信息
        /// </summary>
        /// <param name="keyword">keyword</param>
        /// <returns>string</returns>
        string getWSMsg(string keyword);

        /// <summary>
        /// 根据算法代码检查该参数是否由关键字，该关键字是否来自t_s_param
        /// </summary>
        /// <param name="code">算法代码</param>
        /// <returns>True or False</returns>
        string hasKeyWord(string code);

        /// <summary>
        /// 根据算法代码查找算法状态（是否被审核）
        /// </summary>
        /// <param name="code">算法代码</param>
        /// <returns>状态 0 或者 1</returns>
        string checkAlgoStatus(string code);

        /// <summary>
        /// 检测该函数是否为新算法中的函数
        /// </summary>
        /// <param name="funName">函数名</param>
        /// <returns>True,False</returns>
        string checkFunName(string funName);

        /// <summary>
        /// 解析当前行数据，找出各个函数字符串
        /// </summary>
        /// <param name="funData">行数据</param>
        /// <returns>函数集合</returns>
        string getFunList(string funData);

        /// <summary>
        /// 根据参数名，判断该参数是否有明细
        /// </summary>
        /// <param name="paramName">参数中文名</param>
        /// <returns>是否有明细</returns>
        string paramHasDetail(string paramName);

        /// <summary>
        /// 根据参数中文名获取参数code
        /// </summary>
        /// <param name="paramName">参数中文名</param>
        /// <returns>参数code</returns>
        string getParamCodeByParamName(string paramName);

        
        /// <summary>
        /// 获取所有的公式
        /// </summary>
        /// <returns>公式对象集合</returns> 
        List<AdvAlgo> getAllAchivmentAlgo();

        /// <summary>
        /// 根据公式代码获取公式详细数据
        /// </summary>
        /// <param name="code">公式代码</param>
        /// <returns>公式对象</returns>
        AdvAlgo getAchivmentAlgoByCode(string code);

    }
}

File: AMS_算法公式组件/code/server/source/com.yss.ams.visaval.support/src/main/java/com/yss/ams/visaval/support/service/IVisAdvAlgoService.java
package com.yss.ams.visaval.support.service;

import java.util.List;
import java.util.Map;

import com.yss.ams.visaval.support.pojo.AdvAlgo;
import com.yss.ams.visaval.support.pojo.AdvAlgoPara;
import com.yss.ams.visaval.support.util.pojo.Parameter;
import com.yss.framework.api.common.co.BasePojo;
import com.yss.framework.api.exception.YssException;
import com.yss.framework.api.mvc.biz.IServiceBus;

public interface IVisAdvAlgoService extends IServiceBus{
	public String getAllDate() throws YssException ;
	public String checkFormula(String isNew,String targetStr) throws YssException ;
	public String getFunc(String funType ) throws YssException;
	public String getRealExpression(String customize) throws YssException;
	public String getKeyWords(String keyType) throws YssException;
	public List<Parameter> getAllParameters();
	public AdvAlgo getAdvAlgo(String key);
	
	/**
	 * 获取A区树所有数据
	 * @return
	 */
	public String getAllDataTree_A();
	
	/**
	 *  算法界面优化  树加载
	 *  @author 马向峰   STORY #31713 【产品优化】算法公式配置优化
	 *  @Date 20170713
	 * @return	List<AlgoAPI>
	 */
	public String getTreeData(String algo_type);
	
	/**
	 * 算法界面，根据树节点代码查找参数和返回值
	 * @author 马向峰   STORY #31713 【产品优化】算法公式配置优化
	 *  @Date 20170713
	 * @param code	节点代码
	 * @return	DataAPI  参数和返回值的包装类
	 */
	public String getDataAPIByCode(String code);
	
	public Map<String,String> getParasByCode(String code);
	
	/**
	 * 参考算法
	 * @return
	 */
	public String getAlgos();
	
	public String getAlgoDesc(String code);
	
	/**
	 * 根据算法代码查找算法的所有信息,内包含算法的中英文形式（若中文字段有值则表示新算法，否则为旧算法）
	 * @param code
	 * @return
	 */
	public BasePojo getAlgoByCode(String code);
	
	/**
	 * 根据函数代码查询参数显示规则
	 * @param code
	 * @return
	 */
	public Map<String,Boolean> getRuleByFunCode(String code);
	
	/**
	 * 根据正则表达式解析合法的函数表达式
	 * @param fun
	 * @return
	 */
	public List<String> FunRegex(String fun);
	
	/**
	 * 处理算法预览
	 * @param mapStr 多个下拉框信息
	 * @param msg 算法信息
	 * @param oneSel 单个下拉框信息
	 * @return
	 */
	public String preview(String msg);
	
	/**
	 * 判断新旧算法：True(新算法),False(旧算法)
	 * @param code
	 * @return
	 */
	public String isNewAlgo(String code);
	
	/**
	 * 检查该参数是否由关键字
	 * @param code
	 * @return
	 */
	public String hasKeyWord(String code);
	
	/**
	 * 根据算法代码查找算法状态是否被审核
	 * @param code
	 * @return
	 */
	public String checkAlgoStatus(String code);
	
	/**
	 * 检查该函数是否为新算法的函数
	 * @param funName
	 * @return
	 */
	public String checkFunName(String funName);
	
	/**
	 * 根据函数信息，获取每个单个函数字符串
	 * @param funData
	 * @return
	 */
	public String getFunList(String funData);
	
	/**
	 * 根据参数关键字以及参数中文值去获取参数英文
	 * 如果在上下文中找不到就从参数数据表中找然后在放入上下文中供下次使用
	 * 上下文和参数数据表都没有找到返回空值
	 * @param keyword 参数关键字
	 * @param name    参数中文
	 * @return
	 */
	public String getCnToEnCode(String keyword,String name);
	
	/**
	 * 根据参数的中文名判断该参数是否有明细
	 * @param paramName_CH
	 * @return
	 */
	public String paramHasDetail(String paramName_CH);
	
	/**
	 * 根据参数中文名获取参数code
	 * @param paramName_CH
	 * @return
	 */
	public String getParamCodeByParamName(String paramName_CH);
	
	
	 /**
	  * @Title getAllAchivmentAgio 
	  * @Description 获取所有的业绩提成算法公式
	  * @author lff
	  * @date 2019年3月3日上午11:02:49
	  * @throws Exception
	  * @return List<BasePojo>
	  */
	 public List<AdvAlgo>  getAllAchivmentAlgo()throws Exception;
    
	 /**
	  * @Title getAchivmentAgioByCode 
	  * @Description 根据公式代码获取公式详细数据
	  * @author lff
	  * @date 2019年3月3日下午2:30:45
	  * @param code
	  * @return
	  * @throws Exception
	  * @return AdvAlgo
	  */
	 public AdvAlgo getAchivmentAlgoByCode(String code) throws Exception;
	 
	 /**
	  * 
	  * @Title getParamListByCode 
	  * @Description 根据公式查询出对应的参数信息
	  * @author yinyuyi
	  * @date 2019年7月5日下午3:40:23
	  * @param paramCodeList
	  * @return
	  * @throws Exception
	  * @return List<AdvAlgoPara>
	  */
	 public List<AdvAlgoPara> getParamListByCode(List<String> paramCodeList) throws Exception;
}
